### PR TITLE
Add explicit SSL version for Python 2.7.x

### PR DESCRIPTION
**Pull request by brendandburns**
Addresses https://github.com/Azure/azure-cli/issues/2525

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -236,6 +237,17 @@ def acs_install_cli(resource_group, name, install_location=None, client_version=
     else:
         raise CLIError('Unsupported orchestrator type {} for install-cli'.format(orchestrator_type))
 
+def _ssl_context():
+    if sys.version_info < (2, 8):
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: `ssl. create_default_context` is new in `3.4`. We should cap the version at `3.4`. It is uncertain if the API will be available in `2.8`.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -236,6 +237,17 @@ def acs_install_cli(resource_group, name, install_location=None, client_version=
...
+        return ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    else:
+        return ssl.create_default_context()
+
+def _urlretrieve(url, filename):
+    req = urlopen(url, context=_ssl_context())
+    out = open(filename, "wb")
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Use a `with` structure?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: done.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: done.

**Comment by brendandburns** [comment]
Comments addressed. Thanks!

**Pull request merged**